science/nektarpp/Portfile

```diff
@@ -0,0 +1,163 @@
+                        3D element support, hybrid shaped elements and continuous and \
+                        discontinuous implementations. As well as a modern C++ API, it \
+                        ships with a range of solvers including CFD, advection-diffusion \
+                        equations and cardiac electrophysiology.
+
+# Additional dependencies for Tetgen and Triangle
+master_sites            ${master_sites}:main \
+                        ${homepage}thirdparty/:thirdparty
+
+set main_distfile       ${distfiles}
+distfiles               ${main_distfile}:main \
+                        tetgen-1.5.zip:thirdparty \
+                        triangle-1.6.zip:thirdparty
+
+checksums               nektar-5.0.0.tar.bz2 \
+                        rmd160  f588a4d6429b1ffc85be538885cbb09886ac9cba \
+                        sha256  d98c87af64e6d5cce04f8f60045a63fe63a79274c72a0ba368716fe603a4b625 \
+                        size    48486184 \
+                        tetgen-1.5.zip \
+                        rmd160  d8785f4ca6b26608ec9423f7574a0e736380370a \
+                        sha256  52207793198746de14abcb30a0aeed617d7348ff37544e7d7e65aaaa76d7fa70 \
+                        size    264128 \
+                        triangle-1.6.zip \
+                        rmd160  c92c011c442157cae5abcf246935d61bdfeede26 \
+                        sha256  7d74d7efbe35dc5a497bd179bdc87b4670740e466f67043d41e0b0546ce76cb5 \
+                        size    130331
+
+# Only the full Nektar++ source should be extracted - CMake will extract any
+# third-party dependencies.
+extract.only            ${main_distfile}
+
+patchfiles              no-homebrew.patch \
+                        avoid-python-install.patch \
+                        thirdparty-sources.patch
+
+compiler.cxx_standard   2011
+cmake.build_type        Release
+
+mpi.setup
+
+depends_lib-append      port:arpack \
+                        port:boost \
+                        port:fftw-3 \
+                        port:opencascade \
+                        port:scotch \
+                        port:tinyxml \
+                        port:zlib
+
+# Don't build demos and tests by default.
+configure.args-append   -DNEKTAR_BUILD_DEMOS=OFF \
+                        -DNEKTAR_BUILD_TESTS=OFF \
+                        -DNEKTAR_BUILD_UNIT_TESTS=OFF \
+                        -DNEKTAR_SOLVER_DIFFUSION=OFF \
+                        -DNEKTAR_SOLVER_DUMMY=OFF \
+                        -DNEKTAR_SOLVER_ELASTICITY=OFF \
+                        -DNEKTAR_SOLVER_MMF=OFF \
+                        -DNEKTAR_USE_ARPACK=ON \
+                        -DNEKTAR_USE_FFTW=ON \
+                        -DNEKTAR_USE_MESHGEN=ON \
+                        -DNEKTAR_USE_SCOTCH=ON
+
+variant demos description {
+    Build additional Nektar++ demos
+} {
+    configure.args-replace \
+                        -DNEKTAR_BUILD_DEMOS=OFF -DNEKTAR_BUILD_DEMOS=ON
+}
```

Comment:
Does it really take such a long time to build the demos that a variant is needed and this is not always done?

Reply:
No, not really -- it just produces rather a lot of additional executables that aren't really all that interesting unless you specifically want them.

Reply:
@mf2k I decided this is not really interesting enough to warrant an entire variant, so I've removed it. The demos are quite specialised, for the most part, so if necessary one can compile the demo from source against the library in any case. Let me know if you want any other changes.